core/src/main/java/com/google/instrumentation/stats/ContextUtils.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import com.google.instrumentation.common.NonThrowingCloseable;
+import io.grpc.Context;
+
+/**
+ * Util methods/functionality to interact with the {@link io.grpc.Context}.
+ *
+ * <p>Users must interact with the current Context via the public APIs in {@link
+ * StatsContextFactory} and avoid usages of the {@link #STATS_CONTEXT_KEY} directly.
+ */
+public final class ContextUtils {
+
+  /**
+   * The {@link io.grpc.Context.Key} used to interact with {@link io.grpc.Context}.
+   */
+  public static final Context.Key<StatsContext> STATS_CONTEXT_KEY = Context.key(
+      "instrumentation-stats-key");
+
+  // Static class.
+  private ContextUtils() {
+  }
+
+  /**
+   * Returns The {@link StatsContext} from the current context.
+   *
+   * @return The {@code StatsContext} from the current context.
+   */
+  static StatsContext getCurrentStatsContext() {
+    return STATS_CONTEXT_KEY.get(Context.current());
+  }
+
+  /**
+   * Construct a new {@link Context} where the given {@link StatsContext} is set, with current
+   * context as its parent.
+   *
+   * @param statsContext The {@code StatsContext} to be set to the current context.
+   * @return a new context where the given {@code StatsContext}, and its parent is current context.
+   */
+  static Context withStatsContext(StatsContext statsContext) {
```

Comment:
I'm considering exposing this as a public API in `StatsContextFactory`, because compared to `withScopedStatsContext()` (which internally calls `Context.attach()` and `detach()`), this method gives user more flexibility on using the created `Context`.